app/src/main/java/com/github/libretube/ui/fragments/WatchHistoryFragment.kt

```diff
@@ -114,26 +114,15 @@ class WatchHistoryFragment : DynamicLayoutManagerFragment(R.layout.fragment_watc
                 .show()
         }
 
-        binding.filterTypeTV.setOnClickListener {
-            val filterOptions = resources.getStringArray(R.array.filterOptions)
-
-            BaseBottomSheet().apply {
-                setSimpleItems(filterOptions.toList()) { index ->
-                    binding.filterTypeTV.text = filterOptions[index]
-                    viewModel.selectedTypeFilter = index
-                }
-            }.show(childFragmentManager)
-        }
-
-        binding.filterStatusTV.setOnClickListener {
-            val filterOptions = resources.getStringArray(R.array.filterStatusOptions)
-
-            BaseBottomSheet().apply {
-                setSimpleItems(filterOptions.toList()) { index ->
-                    binding.filterStatusTV.text = filterOptions[index]
-                    viewModel.selectedStatusFilter = index
-                }
-            }.show(childFragmentManager)
+        binding.statusFilterChips.setOnCheckedStateChangeListener { _, checkedIds ->
+            val continueWatchingEnabled = checkedIds.contains(binding.chipContinue.id)
+            val finishedEnabled = checkedIds.contains(binding.chipFinished.id)
+            viewModel.selectedStatusFilter = when {
+                continueWatchingEnabled && finishedEnabled -> 0
+                continueWatchingEnabled -> 1
+                finishedEnabled -> 2
+                else -> 0
+            }
```

Comment:
The new code no longer allows deselecting all filter chips. Not sure if that's something that should be possible, but in that case there should be some feedback to the user (maybe disabling the other chip?).

Reply:
> The new code no longer allows deselecting all filter chips
>
Well, if you deselect all, you would have 0 results, right? So this doesn't make much sense imo.